Stop search has failures matching street intersections in certain cases

jeffremer opened the pull request:
Example:

2.0.0p0 :017 > r21.inbound.stop_at("3520")
 => #<Muni::Stop tag="3520", title="26th St & Noe St", lat="37.7482699", lon="37.7482699", stopId="37.7482699", route_tag="24", direction="24_IB1"> 

2.0.0p0 :015 > r21.inbound.stop_at("26th St & Noe St")
 => nil

jeffremer commented:
HI @gcapiel, thanks for the report. Assuming you are using the 21 Hayes (tag 21), it doesn't appear to stop at a stop tagged "3520". Correct me if I am wrong. But these look like the available inbound stops for the 21.

Given: `route21 = Muni::Route.find(21)`

``` ruby
irb(main):018:0> route21.inbound.stops.map(&:tag)
=> ["7499", "7561", "4988", "5002", "4986", "5000", "4980", "4992", "5004", "5012", "4994", "5015", "4984", "5260", "4921", "4920", "4923", "7463", "5652", "5651", "5650", "5647", "5645", "5643", "5640", "5685", "7264", "5658", "6475", "36497"]
```

Or as titles:

``` ruby
irb(main):017:0> route21.inbound.stops.map(&:title)
=> ["Fulton St & Shrader St", "Hayes St & Shrader St", "Hayes St & Clayton St", "Hayes St & Masonic Ave", "Hayes St & Central Ave", "Hayes St & Lyon St", "Hayes St & Baker St", "Hayes St & Divisadero St", "Hayes St & Pierce St", "Hayes St & Steiner St", "Hayes St & Fillmore St", "Hayes St & Webster St", "Hayes St & Buchanan St", "Laguna St & Hayes St", "Grove St & Laguna St", "Grove St & Gough St", "Grove St & Van Ness Ave", "Grove St & Polk St", "Market St & 9th St", "Market St & 8th St", "Market St & 7th St", "Market St & 6th St", "Market St & 5th St", "Market St & 4th St", "Market St & 3rd St", "Market St & New Montgomery St", "Market St & 1st St", "Market St & Beale St", "Spear St & Market St", "Steuart St & Mission St"]
```

In that case I'd expect those calls to return nil. It's been a while since I've taken the 21, but from what I remember it doesn't stop near 26th and Noe.

If you're trying to get predictions for a another line do you mind provide a full failing test case?

gcapiel commented:
Sorry, I should have used a better variable name.  I did this first:

r = Muni::Route.find(24)

then 
2.0.0p0 :022 > r.inbound.stop_at("3520")
 => #<Muni::Stop tag="3520", title="26th St & Noe St", lat="37.7482699", lon="37.7482699", stopId="37.7482699", route_tag="24", direction="24_IB1"> 

2.0.0p0 :015 > r.inbound.stop_at("26th St & Noe St")
=> nil

which should not be the case.

jeffremer commented:
Gotcha, I can reproduce it now - I'll take a look.

jeffremer committed:
Properly finds stops with numeric titles

This fixes an issue where stops with titles that started with numbers
were being confused with stop tags.

Fixes #9

jeffremer committed:
Fixes VCR/WebMock version warnings

jeffremer committed:
Bumps version to 0.0.6

jeffremer commented on lib/muni/direction.rb:
```diff
@@ -3,11 +3,18 @@
   class Direction < Base
     def stop_at(place)
-      return stops.select{ |stop| stop.tag == place }.first if place =~ /[1-9][0-9]+/
+      if stop = stops.detect { |stop| stop.tag == place }
```
@gcapiel, this is the actual fix for your issue.

jeffremer commented on lib/muni/direction.rb:
```diff
@@ -3,11 +3,18 @@
+        return stop
+      end
+
+      if stop = stops.detect { |stop| stop.title == place }
```
This is just a nice little shortcut to reduce the `Amatch` overhead.

jeffremer commented:
@gcapiel, this pull request fixes your issue. If you care to take a look feel free to comment. Otherwise I'll merge and cut a version to push to ruby gems in a bit.

gcapiel commented:
@jeffremer that fixed it.  Thanks!  I'm using it to create my own bus stop sign for my house connected to my Raspberry Pi.  Check it out:

![Screen Shot 2013-04-13 at 5 40 54 PM](https://f.cloud.github.com/assets/99528/377337/0b63ca1a-a49c-11e2-817f-4aec03f5bc90.png)

jeffremer commented:
Nice! That's a fantastic project. I've been meaning to get my hands on a Raspberry Pi.

If you care to document the project (Instructables?) I'd love to add a link in the `README.md` for example uses of the gem.
